.github/workflows/e2e-test.yml

```diff
@@ -27,26 +27,77 @@ jobs:
       - name: Test E2E
         run: |
           make test-e2e
-        if: env.GIT_DIFF
+        if: env.GIT_DIFF   
+  test-files-matrix:
+    runs-on: ubuntu-latest
+    outputs:
+      test_matrix: ${{ steps.set_matrix.outputs.test_matrix }}
+    steps:      
+      - uses: actions/checkout@v4
+      - name: Set up test file matrix to run nix tests in parallel
+        id: set_matrix
+        run: |
+          cd ./tests/nix_tests || exit 1
+
+          # Define parallel test files
+          PARALLEL_TESTS=(
+            "test_account.py"
+            "test_filters.py"
+            "test_fee_history.py"
+            "test_grpc_only.py"
+            "test_ibc.py"
+            "test_no_abci_resp.py"
+            "test_osmosis_outpost.py"
+            "test_precompiles.py"
+            "test_priority.py"
+            "test_pruned_node.py"
+            "test_rollback.py"
+            "test_stride_outpost.py"
+            "test_storage_proof.py"
+            "test_zero_fee.py"
+          )
```

Comment:
what would be the reason we don't want to run a specific test file in parallel with the others? 🤔 can't we just use the `test_*.py` pattern to select all test files or are there some escaped atm?

Reply:
the remaining tests are all using the same chain, so it spins up the chain and runs all the tests against it. For the other cases (the files in the list), a chain with specific configuration is setup